### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by annq3012**
## Description
Create Data Table By Migration Laravel
## Reference Requirements
- Product Backlog ID: PBI-0
- Task ID: #0.3

## Migrations
YES 

## Impacted Areas in Application
List general components of the application that this PR will affect:

-

## SQL query and `explain`
List of all queries and attach the result of each explaining query (in text, screenshot...)

- Refer to [Using Explain with MySQL](https://viblo.asia/p/su-dung-explain-de-toi-uu-cau-lenh-mysql-BYjv44gmvxpV)

### CODE REVIEW

File: database/migrations/2017_12_30_034453_create_books_table.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            $table->increments('id');
+            $table->integer('category_id')->unsigned();
+            $table->foreign('category_id')
+                  ->references('id')->on('categories')
+                  ->onUpdate('cascade')
+                  ->onDelete('cascade');
+            $table->string('name');
+            $table->string('author');
+            $table->string('year');            
+            $table->string('description');
```
Comment (at-anhngo) [inline database/migrations/2017_12_30_034453_create_books_table.php]: books table does not have price column

File: database/migrations/2017_12_30_034453_create_books_table.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            $table->string('donate_by');
+            $table->string('image');
+            $table->float('avg_rating');
+            $table->integer('total_rating');
```
Comment (at-anhngo) [inline database/migrations/2017_12_30_034453_create_books_table.php]: avg_rating and integer must have default value
--> default value = 0

File: database/migrations/2017_12_30_034453_create_books_table.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            $table->string('name');
+            $table->string('author');
+            $table->string('year');            
+            $table->string('price');            
+            $table->string('description');
+            $table->string('donate_by');
+            $table->string('image');
+            $table->float('avg_rating');
+            $table->integer('total_rating');
+            $table->tinyInteger('status');
```
Comment (at-anhngo) [inline database/migrations/2017_12_30_034453_create_books_table.php]: status must have default value
--> default value = 1

File: database/migrations/2017_12_30_034453_create_books_table.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                  ->onUpdate('cascade')
+                  ->onDelete('cascade');
+            $table->string('name');
+            $table->string('author');
+            $table->string('year');            
+            $table->string('price');            
+            $table->string('description');
+            $table->string('donate_by');
+            $table->string('image');
+            $table->float('avg_rating')->defautl(0.0);
```
Comment (at-anhngo) [inline database/migrations/2017_12_30_034453_create_books_table.php]: incorrect function
-->defautl(0.0) => default(0.0)

File: database/migrations/2017_12_30_034315_create_categories_table.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+class CreateCategoriesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('categories', function (Blueprint $table) {
```
Comment (at-xinhnguyen) [inline database/migrations/2017_12_30_034315_create_categories_table.php]: please change the table name to the singular

File: database/migrations/2014_10_12_000000_create_user_table.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public function up()
+    {
+        Schema::create('user', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('employee_code')->unique();
+            $table->string('name');
+            $table->string('email')->unique();
+            $table->string('team');
+            $table->string('avatar_url')->nullable();
+            $table->tinyInteger('role')->default(0);
```
Comment (at-duyho) [inline database/migrations/2014_10_12_000000_create_user_table.php]: - you should add comment for role.
ex: $table->tinyInteger('role')->default(0)->comment="0: admin; 1:user"

File: database/migrations/2014_10_12_000000_create_user_table.php [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+class CreateUserTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('user', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2014_10_12_000000_create_user_table.php]: - table name have to a plural noun

File: database/migrations/2017_12_30_034315_create_category_table.php [hunk 8]
```diff
@@ -13,13 +13,11 @@ class CreateUsersTable extends Migration
      */
     public function up()
     {
-        Schema::create('users', function (Blueprint $table) {
+        Schema::create('category', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2017_12_30_034315_create_category_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_043417_create_donator_table.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+class CreateDonatorTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('donator', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_043417_create_donator_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_065758_create_book_table.php [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+class CreateBookTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('book', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065758_create_book_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_065758_create_book_table.php [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+                    ->onDelete('cascade');
+            $table->string('name');
+            $table->string('author');
+            $table->string('year');            
+            $table->string('price');            
+            $table->string('description');
+            $table->string('image');
+            $table->float('avg_rating')->defautl(0.0);
+            $table->integer('total_rating')->default(0);
+            $table->tinyInteger('status')->default(1);
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065758_create_book_table.php]: - you should add comment for column

File: database/migrations/2018_01_02_065832_create_borrow_table.php [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+class CreateBorrowTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('borrow', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065832_create_borrow_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_065902_create_post_table.php [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+class CreatePostTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('post', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065902_create_post_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_065902_create_post_table.php [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+            $table->foreign('book_id')
+                  ->references('id')->on('book')
+                  ->onUpdate('cascade')
+                  ->onDelete('cascade');
+            $table->integer('user_id')->unsigned();
+            $table->foreign('user_id')
+                  ->references('id')->on('user')
+                  ->onUpdate('cascade')
+                  ->onDelete('cascade');
+            $table->tinyInteger('type');
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065902_create_post_table.php]: - add comment for column

File: database/migrations/2018_01_02_065930_create_comment_table.php [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+class CreateCommentTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('comment', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_065930_create_comment_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_070003_create_like_table.php [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+class CreateLikeTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('like', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_070003_create_like_table.php]: - table name have to a plural noun

File: database/migrations/2018_01_02_081818_create_rating_table.php [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+class CreateRatingTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('rating', function (Blueprint $table) {
```
Comment (at-duyho) [inline database/migrations/2018_01_02_081818_create_rating_table.php]: - table name have to be a plural noun

**Review by ghost — CHANGES_REQUESTED** [review]
check my comment.

File: database/migrations/2018_01_02_065758_create_book_table.php [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+                    ->references('id')->on('donator')
+                    ->onUpdate('cascade')
+                    ->onDelete('cascade');
+            $table->string('name');
+            $table->string('author');
+            $table->string('year');            
+            $table->string('price');            
+            $table->string('description');
+            $table->string('image');
+            $table->float('avg_rating')->defautl(0.0);
```
Comment (at-anhngo) [inline database/migrations/2018_01_02_065758_create_book_table.php]: default, not defautl

File: database/migrations/2017_12_30_034315_create_categories_table.php [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+            $table->increments('id');
+            $table->string('name');
+            $table->timestamps();
+            $table->softDeletes();
```
Comment (at-xinhnguyen) [inline database/migrations/2017_12_30_034315_create_categories_table.php]: a nghĩ Category k cần softDeletes đâu

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**